### PR TITLE
Block editor: iframe for block based themes (PHP backport)

### PR DESCRIPTION
**Pull request by ellatrix**
Trac ticket: https://core.trac.wordpress.org/ticket/57549
Gutenberg PR: https://github.com/WordPress/gutenberg/pull/46212

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

File: src/wp-includes/block-editor.php [hunk 1]
```diff
@@ -513,6 +513,7 @@ function get_block_editor_settings( array $custom_settings, $block_editor_contex
 	}
 
 	$editor_settings['__unstableResolvedAssets']         = _wp_get_iframed_editor_assets();
+	$editor_settings['__unstableIsBlockBasedTheme']      = wp_is_block_theme();
```
Comment (hellofromtonya) [inline src/wp-includes/block-editor.php:516]: @ellatrix is this a new editor setting? If no, why "unstable"? Can if it stabilized (ie to avoid BC breaks later) for introduction into Core?

Comment (ellatrix) [reply src/wp-includes/block-editor.php:516]: I have no intention of stabilising it, it should just be for internal use. Currently it's just added for the purpose of iframing for block based theses, but in the future iframing should happen for all themes. So this setting will go away. Should I prefix it with `__internal`? Or what would you suggest?

Comment (hellofromtonya) [reply src/wp-includes/block-editor.php:516]: Thanks for sharing more context @ellatrix.

tl;dr
Its name is okay as is.

In this specific use case, this new setting's intent is for internal-only, not for use by extenders, and probably for removal in the future. Looking at the Gutenberg docs, `unstable` is the right naming convention to align to this setting's intent.

**Review by hellofromtonya — APPROVED** [review]
The setting addition matches the change from https://github.com/WordPress/gutenberg/pull/46212/files#diff-6207ce64dd0d1f385378631f2127d8ee06ff954ca0e947d18e3d03cec4f6abe9. This is the complete PHP backport/sync 👍 

This is ready for `commit`.

**Comment by hellofromtonya** [comment]
Committed via changeset https://core.trac.wordpress.org/changeset/55147.
